blockchain/difficulty.go

```diff
@@ -540,6 +540,58 @@ func (b *BlockChain) sumPurchasedTickets(startNode *blockNode, numToSum int64) i
 	return numPurchased
 }
 
+// immatureTickets returns the total number of immature tickets at the
+// current tip and a map of immature ticket counts keyed by the block height at
+// which they were bought at.
+func (b *BlockChain) immatureTickets() (int64, map[int64]int64) {
+	tip := b.bestChain.Tip()
+	var numImmature int64
+	immatureSet := make(map[int64]int64, b.chainParams.TicketMaturity)
+	for node, numTraversed := tip, int64(0); node != nil &&
+		numTraversed <= int64(b.chainParams.TicketMaturity); numTraversed++ {
+		freshStake := int64(node.freshStake)
+		numImmature += freshStake
+		immatureSet[node.height] = freshStake
+		node = node.parent
+	}
+
+	return numImmature, immatureSet
+}
+
+// LiveTicketsAtNextMaturity returns the total number of live tickets
+// available after the ticket maturity period from the current tip.
+// This assumes all blocks mined during the period are voted on
+// by the maximum number of tickets allowed per block.
+func (b *BlockChain) LiveTicketsAtNextMaturity() uint64 {
+	tip := b.bestChain.Tip()
+	b.chainLock.RLock()
+	liveTickets := int64(len(tip.stakeNode.LiveTickets()))
```

Comment:
`PoolSize()` appears to give the same result in less operations, though requesting tickets for a new block template is pretty infrequent.